addons/mail/static/src/core/public_web/thread_model_patch.js

```diff
@@ -0,0 +1,42 @@
+import { patch } from "@web/core/utils/patch";
+import { Thread } from "@mail/core/common/thread_model";
+
+/** @type {import("models").Thread} */
+const ThreadPatch = {
+    /**
+     * Handle the notification of a new message based on the notification setting of the user.
+     * Thread on mute:
+     * 1. No longer see the unread status: the bold text disappears and the channel name fades out.
+     * 2. Without sound + need action counter.
+     * Thread Notification Type:
+     * All messages:All messages sound + need action counter
+     * Mentions:Only mention sounds + need action counter
+     * Nothing: No sound + need action counter
+     *
+     * @param {import("models").Message} message
+     */
+    notifyMessageToUser(message) {
+        if (this.isCorrespondentOdooBot) {
+            return;
+        }
+        const channel_notifications =
+            this.custom_notifications || this.store.settings.channel_notifications;
+        if (
+            !this.mute_until_dt &&
+            !this.store.settings.mute_until_dt &&
```

Comment:
I think a `thread.isMuted`  property is needed if a thread has multiple mute options. I see some parts of the js code that don't account for the setting [in thread action?](https://github.com/odoo/odoo/blob/saas-17.4/addons/mail/static/src/discuss/core/common/thread_actions.js#L43).

Reply:
This is just a move, out-of-scope of PR as a `[FIX]` but master can clean code (or find a genuine issue that justifies it being a `[FIX]` change too)